test/e2e/nsx_subnet_test.go

```diff
@@ -602,3 +699,570 @@ func SubnetValidate(t *testing.T) {
+	_, err = testData.crdClientset.CrdV1alpha1().VPCNetworkConfigurations().Update(context.TODO(), vpcNetworkConfig, v1.UpdateOptions{})
+	require.NoError(t, err)
+	// Verify ns2 has two shared subnets with correct properties
+	sharedSubnets := verifyMultipleSharedSubnets(t, ns2, 2, "shared-subnet", "shared-subnet-2")
+	log.Info("Multiple shared subnet verification complete", "ns2_subnet1", sharedSubnets[0].Name, "ns2_subnet2", sharedSubnets[1].Name)
+}
+
+// findSubnetByNamePrefix finds a subnet with the given name prefix in a list of subnets
+func findSubnetByNamePrefix(subnets []*v1alpha1.Subnet, namePrefix string) *v1alpha1.Subnet {
+	for _, subnet := range subnets {
+		if strings.HasPrefix(subnet.Name, namePrefix) {
+			return subnet
+		}
+	}
+	return nil
+}
+
+// getOppositeConnectivityState returns the opposite connectivity state
+func getOppositeConnectivityState(currentState v1alpha1.ConnectivityState) string {
+	if currentState == v1alpha1.ConnectivityStateConnected {
+		return "DISCONNECTED"
+	}
+	return "CONNECTED"
+}
+
+// convertToConnectivityState converts a string state to v1alpha1.ConnectivityState
+func convertToConnectivityState(state string) v1alpha1.ConnectivityState {
+	if state == "CONNECTED" {
+		return v1alpha1.ConnectivityStateConnected
+	}
+	return v1alpha1.ConnectivityStateDisconnected
+}
+
+// waitForConnectivityStateUpdate waits for a subnet's connectivity state to update
+func waitForConnectivityStateUpdate(t *testing.T, namespace, subnetName, expectedStateStr string) *v1alpha1.Subnet {
+	expectedState := convertToConnectivityState(expectedStateStr)
+
+	log.Info("Waiting for subnet connectivity state to update", "subnet", subnetName, "expected", expectedStateStr)
+	err := wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, 12*time.Minute, false, func(ctx context.Context) (bool, error) {
+		updatedSubnet, err := testData.crdClientset.CrdV1alpha1().Subnets(namespace).Get(context.TODO(), subnetName, v1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		log.Info("Current connectivity state", "subnet", updatedSubnet.Name, "state", updatedSubnet.Spec.AdvancedConfig.ConnectivityState)
+		return updatedSubnet.Spec.AdvancedConfig.ConnectivityState == expectedState, nil
+	})
+	require.NoError(t, err, "Failed to update connectivity state for subnet %s", subnetName)
+
+	// Get the final updated subnet
+	updatedSubnet, err := testData.crdClientset.CrdV1alpha1().Subnets(namespace).Get(context.TODO(), subnetName, v1.GetOptions{})
+	require.NoError(t, err)
+
+	require.Equal(t, expectedState, updatedSubnet.Spec.AdvancedConfig.ConnectivityState,
+		"Subnet %s connectivity state should be %s, got %s",
+		subnetName, expectedState, updatedSubnet.Spec.AdvancedConfig.ConnectivityState)
+
+	return updatedSubnet
+}
+
+// PrecreatedSharedSubnetPoll tests if nsx-operator subnet_poll can update a shared subnet
+func PrecreatedSharedSubnetPoll(t *testing.T) {
```

Comment:
I'm thinking about if we really want to test Subnet poll or if there is a better way to test it other than waiting for 10+minutes. As before this test is added the e2e test takes around 30mins, I feel 10mins idle time is too long comparing to the overall testing time.

Reply:
This depends on luck,  we don’t necessarily have to wait 10 minutes. Maybe by the time this test runs, 8 minutes have already passed. 
After running it a few times, it’s still tolerable. 
If there are problems, we’ll fix them later.